### PR TITLE
Support for '--interactive' option for 'boss init'

### PR DESCRIPTION
**Pull request by srishanbhattarai**
Deals with #74.

# Changes:
* Added Inquirer package 
* Created `inquiries.py` in `boss/core` to handle user interactions for the CLI
* Minor refactors to `initializer.py` to allow for clean handling of interactive flag

boss-interactive

I've left out validation for any of the fields that we get from the user. What do I need to add?

Open to suggestions about messages and anything else. 😄 

EDIT: Added a better GIF.

### CODE REVIEW

**Comment by kabirbaidhya** [comment]
@srishanbhattarai Great. Thanks. Looks all good :+1: .

Suggestion: can we add few more inputs?
1. SSH `user` [default = `boss`]
2. SSH `port` [default = `22`]
3. Deployment remote base dir `deployment.base_dir` [default = `'~/boss'`]

Check these config [values](https://github.com/kabirbaidhya/boss/blob/master/boss/constants.py#L12-L13)

File: boss/core/inquiries.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    questions = [
+        Text(
+            'project_name',
+            message='Enter your project_name',
+        ),
+
+        Text(
+            'user',
+            message='Enter your username',
```
Comment (kabirbaidhya) [inline boss/core/inquiries.py]: I think, `Enter your SSH username` could be more informative.

**Comment by srishanbhattarai** [comment]
> Deployment remote base dir deployment.base_dir [default = '~/boss']
@kabirbaidhya  Isn't this the 3rd option?

**Comment by kabirbaidhya** [comment]
Yes, I didn't realized it earlier. My bad.

**Comment by kabirbaidhya** [comment]
Looks good. Merging.

**Pull request merged**